packages/client/src/components/ReviewDialog/ReviewDialog.jsx

```diff
@@ -46,46 +72,122 @@ export function ReviewDialog({ initialState, handleClose, user, movieId }) {
         sx={{ backgroundColor: 'backgroundDark', color: 'mainGreen' }}
       >
         <RatingStars
-          rating={formData[0].movie_id === movieId ? formData[0].rating : 0}
+          rating={formData && formData.rating}
+          onChange={(e) => {
+            setFormData({ ...formData, rating: e.target.value });
+          }}
         />
         <DialogContentText sx={{ color: 'mainGreen' }}>
           Please, write your review of the movie
         </DialogContentText>
-        <ReviewContentText
-          autoFocus
-          margin="dense"
-          id="name"
-          label="review"
-          multiline
-          rows={4}
-          fullWidth
-          variant="filled"
-          InputLabelProps={{
-            sx: {
-              color: 'mainGreen',
-              textTransform: 'capitalize',
-            },
-          }}
-          InputProps={{ sx: { color: 'mainGreen' } }}
-          value={
-            formData[0].movie_id === movieId
-              ? formData[0].review_text
-              : 'Leave your first review about this movie'
-          }
-        />
+        {formData ? (
+          <ReviewContentText
+            autoFocus
+            margin="dense"
+            id="name"
+            label="review"
+            multiline
+            rows={4}
+            fullWidth
+            variant="filled"
+            InputLabelProps={{
+              sx: {
+                color: 'mainGreen',
+                textTransform: 'capitalize',
+              },
+            }}
+            InputProps={{ sx: { color: 'mainGreen' } }}
+            defaultValue={formData.review_text}
+            onChange={(e) => {
+              setFormData({ ...formData, review_text: e.target.value });
+            }}
+          />
+        ) : (
+          <ReviewContentText
+            autoFocus
+            margin="dense"
+            id="name"
+            label="review"
+            multiline
+            rows={4}
+            fullWidth
+            variant="filled"
+            InputLabelProps={{
+              sx: {
+                color: 'mainGreen',
+                textTransform: 'capitalize',
+              },
+            }}
+            placeholder="Leave your first review about this movie"
+            InputProps={{ sx: { color: 'mainGreen' } }}
+            onChange={(e) => {
+              setFormData({ ...formData, review_text: e.target.value });
+            }}
+          />
+        )}
```

Comment:
Here the ternary operator "? : " returns two almost identical ReviewContentText components. They only differ by the placeholder and defaultValue props... The code would be cleaner if you could put conditions on this two properties, but inside one component.
something like: 
placeholder = {review ? "" : "Leave your first review about this movie"}
defaultValue={review ? formData.review_text : ""}

Reply:
okey!